docs/config/lua/config/default_cwd.md

```diff
@@ -0,0 +1,47 @@
+# `default_cwd`
+
+Sets the default current working directory used by the initial window. If
+`wezterm start --cwd /some/path` is used to specify the current working
+directory, that will take precedence.
+
+Commands launched using [`SpawnCommand`](../SpawnCommand.md) will use the
+`cwd` specified in the `SpawnCommand`, if any.
+
+Panes/Tabs/Windows created after the first will generally try to resolve the
+current working directory of the current Pane, preferring
+[a value set by OSC 7](../../../shell-integration.markdown) and falling back to
+attempting to lookup the `cwd` of the current process group leader attached to a
+local Pane. If no `cwd` can be resolved, then the `default_cwd` will be used.
+If `default_cwd` is not specified, then the home directory of the user will be
+used.
+
+## Example Flowchart
+
+```text
+                             Is initial window?
+               ______________________|______________________
+              |                                             |
+             Yes                                            No
+              |                                             |
+       Opened with CLI                  Opened with a `SpawnCommand` that includes
+      and `--cwd` flag?                  `cwd` or is a new pane, tab, or window?
+    __________|__________                         __________|__________
+   |                     |                       |                     |
+  Yes                    No                      No                   Yes
+   |                     |                       |                     |
+  Use                    |                Is there a value    Use `cwd` specified
+`--cwd`                  |                 set by OSC 7?       by `SpawnCommand`
+                         |             __________|__________
```

Comment:
That looks great!   Did you render that by hand?

There's a bit missing; If there is no OSC 7 value, then there is a "can the cwd of the process group leader be resolved?" if yes, use that, if no --> "if default_cwd defined".

A footnote describing that could be something like:

> On macOS and Linux, wezterm can attempt to resolve the process group leader and then attempt to resolve its current working directory.  This is not guaranteed to succeed, and there are a number of potential edge cases (which is another reason for configuring your shell to use OSC 7 sequences).

Reply:
Yes, I've made this flow chart by hand. I make a lot of charts like this for work, but not in ascii art fortunately :blush: